Kitodo-API/src/main/java/org/kitodo/config/Config.java

```diff
@@ -11,141 +11,197 @@
+     *            not exist or exception occurred
+     * @return boolean value for the requested key, or {@code defaultIfNull} if
+     *         not found or not parsing
      */
-    public static boolean getBooleanParameter(String parameter, boolean defaultIfNull) {
+    public static boolean getBooleanParameter(String key, boolean defaultValue) {
         try {
-            return getConfig().getBoolean(parameter, defaultIfNull);
+            return getConfig().getBoolean(key, defaultValue);
         } catch (ConversionException e) {
-            return defaultIfNull;
+            logger.catching(e);
+            logger.warn(
+                "Configuration found in kitodo_config.properties for key {} is defined, but cannot be converted to boolean!",
+                key);
+            return defaultValue;
         }
     }
 
     /**
-     * Request int-parameter from Configuration.
+     * Returns the selected int parameter from the configuration file. If no
+     * such parameter exists, or the value cannot be parsed to {@code int},
+     * returns {@code 0}.
      *
-     * @param inParameter
-     *            name of parameter in config file
-     * @return Parameter as int
+     * @param key
+     *            key whose value is to be returned
+     * @return int value for the requested key, or {@code 0} if not found or not
+     *         parsing
      */
-    public static int getIntParameter(String inParameter) {
-        return getIntParameter(inParameter, 0);
+    public static int getIntParameter(String key) {
+        return getIntParameter(key, INT_PARAMETER_NOT_DEFINED_OR_ERRONEOUS);
     }
 
     /**
-     * Request int-parameter from Configuration with default-value.
+     * Returns the selected int parameter from the configuration file. If no
+     * such parameter exists, or the value cannot be parsed to {@code int},
+     * returns the provided default value.
      *
-     * @param inParameter
-     *            name of parameter in config file
-     * @param inDefault
-     *            default value in case parameter taken from config file is null or
-     *            exception occurred
-     * @return Parameter as int
+     * @param parameter
+     *            key whose value is to be returned
+     * @param defaultIfNull
+     *            default value in case parameter taken from config file does
+     *            not exist or exception occurred
+     * @return int value for the requested key, or {@code defaultIfNull} if not
+     *         found or not parsing
      */
-    public static int getIntParameter(String inParameter, int inDefault) {
+    public static int getIntParameter(String key, int defaultValue) {
         try {
-            return getConfig().getInt(inParameter, inDefault);
+            return getConfig().getInt(key, defaultValue);
         } catch (ConversionException e) {
-            return inDefault;
+            logger.catching(e);
```

Comment:
Two different loggers in one catch block?

Reply:
This was there already. I only added logging of caught exception. Because as to coding guidelines, exception being caught must be logged.

Reply:
Is it necessary to double log it? I think one logger on level error is enough.

Reply:
_Warn_ is a high log level, always logged, also in production environments. I think it is not necessary to log a stack trace there, because a configuration problem is reported which can be fixed by reparing the configuration file, and does not need the stack trace.

Reply:
So please use only one logger, if you find catching more suitable use this one and remove warn.